### PR TITLE
DEV-3882 Dashboard graph initial setup and x-axis

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

- Adds the Warnings Information section and the X-axis of the graph
- Breaking this story up into smaller PRs to show incremental progress (since the page is dark released) and for incremental code/design review

**Technical details:**

- Adds some shared visualization styles (will need to be reconciled with styling in #1182 and the table section)
- Initializes a wrapper container that makes an API call when applied filters change
- Adds a temporary loading state (will work with UX to improve the appearance)
- Uses D3's `scaleBand` to evenly space submission time periods (fiscal quarters) along the X-axis
- Includes some preliminary set up for the Y-axis and chart body based on USAspending's [BarChartStacked](https://github.com/fedspendingtransparency/usaspending-website/blob/dev/src/js/components/account/visualizations/time/chart/BarChartStacked.jsx)

**Link to JIRA Ticket:**
[DEV-3882](https://federal-spending-transparency.atlassian.net/browse/DEV-3882) (subtask of [DEV-3665](https://federal-spending-transparency.atlassian.net/browse/DEV-3665))

**Mockup**
https://bahdigital.invisionapp.com/share/7GIABMNX8MV#/296021489_Historical-Search-Monthly-Single-Data

**Testing**
Filters in `dev`: SSA, FY 17, All quarters, Cross: C/D1 should display "FY 17 / Q2" centered along the x-axis

The following are ALL required for the PR to be merged:
- [ ] Frontend review completed
- [x] Demo including Design/Testing/Front-end OR Provided Instructions for Local Testing above and in JIRA
- [x] Design review complete
- [x] Verified cross-browser compatibility
- [x] Link to this PR in JIRA ticket

### CODE REVIEW

File: src/js/components/dashboard/graph/BarChartStacked.jsx [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    componentDidMount() {
+        this.buildVirtualChart(this.props);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.data !== this.props.data) {
+            this.buildVirtualChart(this.props);
+        }
+        else if (prevProps.width !== this.props.width || prevProps.height !== this.props.height) {
+            this.buildVirtualChart(this.props);
```
Comment (alburde1) [inline src/js/components/dashboard/graph/BarChartStacked.jsx]: Why are these two different if statements? You call the same function with the same parameters

File: src/js/components/dashboard/graph/BarChartXAxis.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    render() {
+        const labels = this.props.items.map((item) => (
+            <BarChartXAxisItem
+                {...item}
+                key={item.value} />
+        ));
+
+        return (
+            <g
+                className="bar-axis">
```
Comment (alburde1) [inline src/js/components/dashboard/graph/BarChartXAxis.jsx]: Why are these on newlines if there's only one thing being passed to them? Is that standard or are we expecting more variables?

File: src/js/helpers/util.js [hunk 3]
```diff
@@ -113,6 +113,30 @@ export const convertDateToQuarter = (date) => {
     return quarter;
 };
 
+export const convertQuarterToDate = (qtr, year) => {
```
Comment (alburde1) [inline src/js/helpers/util.js]: Is there a reason we have this?

Comment (ebdabbs) [reply src/js/helpers/util.js]: This was if we wanted the x-axis to reflect time instead of evenly spacing each submission, so it can be removed!

**Pull request merged**